nix/vaultenv-static.nix

```diff
@@ -0,0 +1,89 @@
+# This is the default build script taken from static-haskell-nix.
+# We've adjusted some things and removed some instructions that
+# we didn't have a usecase for.
+#
+# Run using:
+#
+#     $(nix-build --no-link -A full-build-script nix/vaultenv-static.nix)
+#
+# The invocation above will build a buildscript and then run it.
+# (The output of `nix-build --no-link -A full-build-script` is a
+# path to the build script. The `$()` invokes the script.)
+#
+# That will:
+#
+#   - Fetch a stackage snapshot and run stack2nix on it. This
+#     yields a bunch of nix derivations of the entire snapshot.
+#     Some of these will be used in the actual build for the
+#     vaultenv package.
+#   - Download or build a GHC compiled with musl and integer-simple.
+#   - Build vaultenv with integer-simple and link everything
+#     statically. Dependencies are taken from stack.yaml in this
+#     repository.
+#
+# The final output of the invocation above will print a Nix store path
+# like `/nix/store/ak6qm8qmb66zwri7y16sqina1pvn0gmz-vaultenv-real-0.10.0`
+# which will contain a fully static vaultenv binary in the `bin/` subdir.
+{
+  stack2nix-output-path ? "custom-stack2nix-output.nix",
+}:
+let
+  # Field from the hpack/cabal file. We append `-real` to work around
+  # vaultenv also being included in `nixpgks`, which leads to conflicts.
+  cabal-package-name = "vaultenv-real";
+
+  # This has to match the compiler used in the Stackage snapshot.
+  # Update this when the Stackage snapshot changes the version of
+  # GHC it uses.
+  compiler = "ghc865";
+
+  # Pin versions of static-haskell-nix and nixpkgs.
+  static-haskell-nix = import ./static-haskell-nix.nix;
```

Comment:
Perhaps instead of importing the `static-haskell-nix.nix` file:

```nix
sources = import ./sources.nix;
static-haskell-nix = sources.static-haskell-nix;
```

That way we can delete the `static-haskell-nix.nix` file, which right now only contains 4 lines of code :+1:

Reply:
Even cleaner, good suggestion!